Backend/app/controllers/customer.controller.js

```diff
@@ -72,9 +72,22 @@ const customerController = {
     }
   },
   async getUserData(req, res, next) {
-    try {
-      const user_data = await Customer.queryUserData(req.query);
-      res.send(user_data);
+    try { 
+      if (req.user.role_id === 1){ 
+        const user_data = await Customer.queryUserData(
+          {
+            chapter_id: req.user.chapter_id,
+            ...req.query
+          });
+        res.send(user_data);
+      }
+      else if (req.user.role_id === 2){
+        const user_data = await Customer.queryUserData(req.query);
+        res.send(user_data);
+      }
+      else{
+        throw Error;
```

Comment:
Don't throw errors in controllers. You have to pass it to the error handler with `next(new Error('error string here'));`